### PR TITLE
storage: reduce tscache page size in tests

### LINKED ISSUE #20401: tscache: skiplist implementation OOMs stress tests
Something about the new skiplist-based tscache causes our stress tests to quickly OOM on certain packages in certain configurations. See:

* #20394
* #20387 
* #20386
* #20385
* #20384
* #20383
* #20382
* #20381
* #20379 
* #20377

### PR DESCRIPTION
**Pull request by nvb**
Fixes  #20401.
Closes #20405.

We were already reducing the page size of the `sklImpl` timestamp cache
during race testing. We now do it during all testing that uses test
configurations.

### CODE REVIEW

File: pkg/server/node_test.go [hunk 1]
```diff
@@ -94,8 +94,6 @@ func createTestNode(
 	)
 	cfg.DB = client.NewDB(sender, cfg.Clock)
 	cfg.Transport = storage.NewDummyRaftTransport(st)
-	cfg.MetricsSampleInterval = metric.TestSampleInterval
```
Comment (nvanbenschoten) [inline pkg/server/node_test.go]: A bit of this was just driveby cleanup. Here, `TestStoreConfig` already set these. `server.Context` -> `server.Config` must have been missed before.

**Comment by nvanbenschoten** [comment]
Ping.

**Comment by tschottdorf** [comment]
:lgtm:

---

Reviewed 11 of 11 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/server/node_test.go, line 97 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20414#-L-bGPSGIIXHnnnnA3xG:-L-bGPSGIIXHnnnnA3xH:b-vpvapw) ([raw file](https://github.com/cockroachdb/cockroach/blob/692a832615a9929823a57004357f588e402732b0/pkg/server/node_test.go#L97)):*
> ```Go
> 	cfg.DB = client.NewDB(sender, cfg.Clock)
> 	cfg.Transport = storage.NewDummyRaftTransport(st)
> 	cfg.MetricsSampleInterval = metric.TestSampleInterval
> ```

What's this? Just a drive-by?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20414#-:-L-bGapQo3c1Eg_a86Fu:bnfp4nl)*

**Review by benesch — COMMENTED** [review]
Whoops, sorry, I started reviewing this and got stymied by my lack of understanding of the new tscache. Approach LGTM! Deferring to @tschottdorf on the specifics.

**Comment by nvanbenschoten** [comment]
Thanks for the reviews.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/server/node_test.go, line 97 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20414#-L-bGPSGIIXHnnnnA3xG:-L-bWegE6mX7W13bxNPU:b-e8xb4x) ([raw file](https://github.com/cockroachdb/cockroach/blob/692a832615a9929823a57004357f588e402732b0/pkg/server/node_test.go#L97)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

What's this? Just a drive-by?
</blockquote></details>

Yeah, just a drive by. Tried to leave a note in the comments here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20414#-:-L-bWcNLyxwxhWD8VV7E:b-hlc8b9)*
